nexus_keycode/protocols/small.py

```diff
@@ -195,22 +197,22 @@ def _generate_mac_bits(self, secret_key):
 
         MAC is generated using message ID, type code, and body byte.
 
-        :param secret_key: 16-byte secret key, e.g. "\xff" * 16
-        :type secret_key: `str`
+        :param secret_key: 16-byte secret key, e.g. b"\xff" * 16
+        :type secret_key: `bytes`
         :return: bitstream-packed form of the MAC generated using secret_key.
         :rtype: :class:`bitstring`
         """
 
         # the hash is computed over a struct representation of the message,
         # struct { uint32_t message_id (LE); uint8_t message_type, uint8_t body }
-        structed = "".join(
+        structed = ints_to_bytes(
             [
-                chr(self.id_ & 0xFF),
-                chr((self.id_ & 0x0000FF00) >> 8),
-                chr((self.id_ & 0x00FF0000) >> 16),
-                chr(self.id_ >> 24),
-                chr(self.message_type.value),
-                chr(self.body),
+                (self.id_ & 0xFF),
+                ((self.id_ & 0x0000FF00) >> 8),
+                ((self.id_ & 0x00FF0000) >> 16),
+                (self.id_ >> 24),
+                (self.message_type.value),
```

Comment:
Note: We expect the value of each of these bytes to be 0-255, and it does look like `bytes` on Py3 restricts to (0,256) so we are still good there.